posts/inside-rust/2022-07-29-compiler-team-2022-midyear-report.md

```diff
@@ -0,0 +1,798 @@
+<tr>                                                       <td style=color:darkblue>                      </td><td style=color:darkblue>getting-feedback      </td><td style=color:darkblue>                      </td><td style=color:darkblue>                      </td><td style=color:darkblue>getting-feedback      </td><td style=color:darkblue>                      </td><td style=color:darkblue>                      </td><td style=color:darkblue>                      </td><td style=color:darkblue>                      </td><td style=color:darkblue>                      </td><td style=color:darkblue>                      </td><td style=color:darkblue>getting-feedback      </td><td style=color:darkblue>                      </td><td style=color:darkblue>getting-feedback      </td><td style=color:darkblue>                      </td><td style=color:darkblue>                      </td><td style=color:darkblue>                      </td><td style=color:darkblue>                      </td><td style=color:darkblue>                      </td><td style=color:darkblue>                      </td><td style=color:darkblue>                      </td><td style=color:darkblue>                      </td></tr>
+
+</table>
+
+Here are some trends I noted from looking at the table:
+
+The overwhelming majority of our ambitions, 19 out of 22, reported *some*
+amount of progress. That's great, especially given that several of those
+ambitions were mere aspirations that explicitly had no resources attached to
+them at the start of the year.
+
+Less than a quarter of the ambitions, 6 out of 22, said that their overall
+problem would be resolved, or the bulk of the problem addressed, within the next
+six months. Ten others said they predicted the important components of their
+solution would be available within the next two years, so that's a total of 16
+out of 22 that expect some kind of solution within two years.
+
+  * Note: One respondent noted that the two year threshold embedded in the
+    survey was an "odd timeframe." I do not disagree with that remark. The
+    reality is that I was trying to find a happy medium between two extremes:
+    short timeframes don't allow enough time for significant work to be
+    accomplished, while excessively long timeframes (e.g. greater than five
+    years) are very difficult to predict realistically. But the two year
+    time frame was drawn somewhat out of a hat.
+
+Of the nineteen ambitions that reported some progress so far this year, fourteen
+of them (14/19) also said that at the start of the year, they had either *no
+specific goals*, or *no plan* to achieve their goals. Ten of those fourteen say they now
+have a better understanding of the problem, and five of those fourteen say they
+now have a plan.
```

Comment:
```suggestion
Of the nine concrete initiatives (i.e. ambitions with development resources attached) that reported some progress so far this year, four said they had no specific goal at the start of the year, and two said they had no plan to achieve their goal.
Similarly, of the ten aspirations (i.e. ambitions without development resources attached) that reported progress, six had no specific goals, and two said they had no plan to achieve their goals.

&nbsp; | No Specific Goals | No Plan | Other
--------|-------------------|--------|--------
Concrete Initiative | 4 | 2 | 3
Aspiration | 6 | 2 | 2
 
 Four of those six concrete initiatives that lacked either specific goals or a plan say they now
have a better understanding of the problem; two of them say they
now have a plan.

Five of the eight aspirations that lacked either specific goals or a plan say they now better understand the problem, and two now have a plan.

I spell this out in this level of detail because I was *expecting* to see some strong correlation between having development resources and having a plan, at least compared to what the aspirations reported. I think the numbers above show that expectation was incorrect. I'm not yet sure what lesson to draw from this, apart from being happy that so many ambitions report that they have a better understanding of their respective problem.
```